jaeger-core/src/test/java/io/jaegertracing/JaegerSpanTest.java

```diff
@@ -79,56 +80,56 @@ public void testSpanMetrics() {
   public void testSetAndGetBaggageItem() {
     final String service = "SamplerTest";
     final BaggageRestrictionManager mgr = Mockito.mock(DefaultBaggageRestrictionManager.class);
-    tracer = new Tracer.Builder(service)
+    tracer = new JaegerTracer.Builder(service)
             .withReporter(reporter)
             .withSampler(new ConstSampler(true))
             .withClock(clock)
             .withBaggageRestrictionManager(mgr)
             .build();
-    span = (Span) tracer.buildSpan("some-operation").start();
+    jaegerSpan = (JaegerSpan) tracer.buildSpan("some-operation").start();
```

Comment:
is this cast necessary?

Reply:
Actually, we could mark JaegerTracer's `buildSpan()` to return `JaegerTracer.SpanBuilder`, whose `start()` method would return `JaegerSpan`. This way, callers using a concrete `JaegerTracer` won't need to cast anything...

Reply:
Looks like there are other things to fix if we go this route, as `JaegerTracer.SpanBuilder` is not public, so, some tests cannot access it (`NoopMetricsFactoryTest` is the one that complained first).

I'll open an issue to investigate this.